nomenclature/definition.py

```diff
@@ -60,6 +62,59 @@ def validate(self, df: IamDataFrame, dimensions: list = None) -> None:
         """
         validate(self, df, dimensions=dimensions or self.dimensions)
 
+    def check_aggregate(self, df: IamDataFrame, **kwargs) -> None:
+        """Check for consistency of scenario data along the variable hierarchy
+
+        Parameters
+        ----------
+        df : :class:`pyam.IamDataFrame`
+            Scenario data to be checked for consistency along the variable hierarchy.
+        kwargs : Tolerance arguments for comparison of values
+            Passed to :any:`numpy.isclose` via :any:`pyam.IamDataFrame.check_aggregate`.
+
+        Returns
+        -------
+        :class:`pandas.DataFrame` or None
+            Data where a variable and its computed aggregate does not match.
+
+        Raises
+        ------
+        ValueError
+            If the :any:`DataStructureDefinition` does not have a *variable* dimension.
+        """
+        if "variable" not in self.dimensions:
+            raise ValueError("Aggregation check requires 'variable' dimension.")
+
+        lst = []
+
+        with adjust_log_level(level="WARNING"):
+            for code, attr in self.variable.items():
```

Comment:
Would it be more efficient to iterate only over the variables which actually are present in the input dataframe instead of the whole variable list?

Reply:
Good suggestion, implemented.